src/main/java/org/jenkinsci/plugins/docker/workflow/FromFingerprintStep.java

```diff
@@ -87,8 +89,9 @@ public String getToolName() {
             String fromImage = null;
             FilePath dockerfile = workspace.child(step.dockerfile);
             InputStream is = dockerfile.read();
+            BufferedReader r = null;
             try {
-                BufferedReader r = new BufferedReader(new InputStreamReader(is)); // encoding probably irrelevant since image/tag names must be ASCII
+                r = new BufferedReader(new InputStreamReader(is, "ASCII")); // encoding probably irrelevant since image/tag names must be ASCII
```

Comment:
if this code runs on the slave then yes.  if it runs on the master and does reading vi Jenkins workspace abstraction then it needs something a little more (there is no guarantee that the master and slaves (or jobs) use the same settings.  
I think (and may be wrong) that this runs on the master.